refactor(app): gate protocol calibrate and run on pipette offset cal

sfoster1 opened the pull request:
If the calibration overhaul is enabled, the user should not be able to start calibrating their labware or running their protocol until all pipettes used in the protocol have pipette offset calibrations. If they don't, disable those tabs and display the same kind of warning you get if a pipette isn't attached.

## Testing
- Make sure that if you don't have the cal overhaul enabled you never see this (a good way to test this is to enable cal overhaul, make sure you do see it, and then disable cal overhaul and make sure it goes away)
- Test that having a pipette that _isn't_ used in the protocol and doesn't have a calibration doesn't block you


Closes #6627

sfoster1 committed:
refactor(app): add cal information to pipette readiness selectors

Alters the (monster) selector getProtocolPipettesInfo to also include
whether or not a pipette offset calibration exists for the detected
pipette, which will later be used by UI components to determine whether
the protocol is ready to run. These selectors build in the feature flag
query, and will always indicate that calibration is present if the
feature flag for calibration overhaul is turned off (and now show up as
places to change when we look for things to remove along with the
feature flag later).

In addition to protocolPipettesMatch, which operates unchanged, we now
have protocolPipettesCalibrated (similar semantics but for offset
calibration, keeping in mind that the checks are based on the feature
flag) and protocolPipettesReady, which is Match && Calibrated. Ready is
used in the nav selectors to enable gating - since that's the point of
this exercise - but Match is still used in fileinfo, and Calibrated soon
will be as well.

sfoster1 committed:
refactor(app): gate protocol execution on pip offset cal

Based on the selectors in the previous pipette, if the pipettes required
for the protocol are attached but do not have a pipette offset
calibration, do not allow the user to proceed to calibrate and run.

Closes #6627

emilywools commented:
looks good when I reviewed IRL (via hangout) with Seth

sfoster1 committed:
refactor(app): correct disabled reasons for bad calibration

Have a separate disabled reason for pipettes
matching-but-not-calibrated, which in turn removes the last callsite of
the ready selector, so remove it

mcous commented on app/src/components/FileInfo/MissingItemWarning.js:
```diff
@@ -1,36 +1,48 @@
-import cx from 'classnames'
-import { OutlineButton } from '@opentrons/components'
+import {
+  OutlineButton,
```
I think you meant to import [SecondaryBtn](https://s3-us-west-2.amazonaws.com/opentrons-components/edge/index.html#btn) here. OutlineButton is a much older component that doesn't use style props

mcous commented on app/src/components/FileInfo/MissingItemWarning.js:
```diff
@@ -1,36 +1,48 @@
   return (
     <SectionContentHalf className={styles.align_center}>
       <OutlineButton
+        texttransform={TEXT_TRANSFORM_UPPERCASE}
```
This is the default of both `SecondaryBtn` (and `OutlineButton`). It's not doing anything here (and also needs to be `textTransform` if it was going to do anything)

sfoster1 replied:
oh I had it as `textTransform` and the console complained about it, which all makes sense if this is the wrong component and isn't styled properly

mcous commented on app/src/components/FileInfo/MissingItemWarning.js:
```diff
@@ -1,36 +1,48 @@
     <SectionContentHalf className={styles.align_center}>
       <OutlineButton
+        texttransform={TEXT_TRANSFORM_UPPERCASE}
         Component={Link}
```
When this is switched to a `SecondaryBtn`, the prop to do this is called `as`

mcous commented on app/src/components/FileInfo/MissingItemWarning.js:
```diff
@@ -1,36 +1,48 @@
+        // this needs to be as a class because something about making a button
+        // pretending to be a Link means that stuff specified in the css that's
+        // passed to the link can't be overridden with props
+        className={styles.width_auto}
```
`width: auto` is the default width style on `SecondaryBtn`

mcous reviewed (commented):
Just driving by

sfoster1 committed:
use a much better secondary button class

sfoster1 committed:
fix the tooltip on the continue button

Laura-Danielle reviewed (changes requested):
I am blocked from proceeding when I have two pipettes attached with only one calibrated, and I am only using the calibrated pipette in the protocol.

I don't _think_ this should matter, but both the pipettes attached are P1000 GEN2s.

sfoster1 committed:
refactor(app): s/hasOffsetCalibration/needsOffsetCalibration

Change the name and invert the sense of the prop that checks for pipette
calibration.

When the prop is named hasOffsetCalibration, it's a statement of fact
about the state of the system, and overriding it to `true` in some
circumstances makes that not actually true.

When the prop is named needsOffsetCalibration (and is inverted relative
to hasOffsetCalibration), it's a statement about the requirements that
must be met before the protocol can run, and it's a much more natural
fit for a prop that in reality depends on more information than just whether a
calibration exists for the pipette - information like, "is this pipette
actually used in the protocol" or "is the calibration overhaul feature
flag engaged".

This then makes it easier to consider whether a pipette is actually used
in a protocol when deciding whether it needs calibration.

Laura-Danielle reviewed (approved):
lgtm
